lib/util.js

```diff
@@ -738,7 +738,9 @@ function formatSet(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
   value.forEach(function(v) {
     var nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-    var str = formatValue(ctx, v, nextRecurseTimes);
+    var str = ctx.seen.includes(v) ?
+      ctx.stylize('[Circular]', 'special') :
+      formatValue(ctx, v, nextRecurseTimes);
```

Comment:
I don’t know if you tried that, but wouldn’t it be better to do this type of check/early return in `formatValue`, before the `ctx.seen.push()` call (i.e. `if (ctx.seen.includes(v)) return ctx.stylize('[Circular]', 'special');`)?